### ISSUE TITLE
Functional tests can't load optional checkers

### ISSUE THREAD

**Issue by DudeNr33** [PyCQA/pylint#4331]
### Steps to reproduce

Given a functional test with the following files:

_confusing_consecutive_elif.py:_
```Python
# pylint: disable=missing-module-docstring, missing-function-docstring

def check_config(machine, old_conf, new_conf):
    if old_conf:
        if not new_conf:
            machine.disable()
        elif old_conf.value != new_conf.value:
            machine.disable()
            machine.enable(new_conf.value)
    elif new_conf:  # [confusing-consecutive-elif]
        machine.enable(new_conf.value)

```

_confusing_consecutive_elif.rc:_
```
[MASTER]
load-plugins=pylint.extensions.confusing_elif

```

_confusing_consecutive_elif.txt:_
```
confusing-consecutive-elif:10:4:check_config:Consecutive elif with differing indentation level, consider creating a function to separate the inner elif

```

### Current behavior

The tests fail, apparently because plugins configured via the ``.rc`` file are not loaded:

```
E       AssertionError: Wrong results for file "confusing_consecutive_elif":
E       
E       Expected in testdata:
E         10: confusing-consecutive-elif
```

### Expected behavior
Functional test passes.

### pylint --version output

Result of `pylint --version` output:

```
% pylint --version
pylint 2.8.0.dev1
astroid 2.5.2
Python 3.8.2 (default, Dec 21 2020, 15:06:03) 
[Clang 12.0.0 (clang-1200.0.32.29)]
```

### Proposed fix:
The issue can be resolved if the logic of plugin-loading from the ``Run`` class is copied to the ``LintModuleTest`` class - only three lines have to be added:

``` Python
class LintModuleTest:
    maxDiff = None

    def __init__(self, test_file: FunctionalTestFile, config: Optional[Config] = None):
        _test_reporter = FunctionalTestReporter()
        self._linter = PyLinter()
        self._linter.set_reporter(_test_reporter)
        self._linter.config.persistent = 0
        checkers.initialize(self._linter)
        self._linter.disable("suppressed-message")
        self._linter.disable("locally-disabled")
        self._linter.disable("useless-suppression")
        try:
            self._linter.read_config_file(test_file.option_file)
            if self._linter.cfgfile_parser.has_option("MASTER", "load-plugins"):  # ADDED
                plugins = utils._splitstrip(self._linter.cfgfile_parser.get("MASTER", "load-plugins"))  # ADDED
                self._linter.load_plugin_modules(plugins)  # ADDED
            self._linter.load_config_file()
        except NoFileError:
            pass
        self._test_file = test_file
        self._config = config
```

### Impact Analysis
I have made the change in my local repository and reran the functional tests.
The test ``fixture_docparams_missing`` is now failing.
This seems to be not a regression but a previously false negative test, since the ``.txt`` file with the expected messages is missing. This test also tried to use the ``load-plugins`` option in its ``.rc`` file.
To keep the test suite passing, this functional test should be fixed together with the changed to ``LintModuleTest``.

**Comment by Pierre-Sassoulas** [comment]
Hey, good job and amazing that you fixed another problem with this ! You can update the .txt file with the command given in the fail message :)

**Comment by cdce8p** [comment]
@DudeNr33 Thanks for taking the time to investigate this issue! When you open the PR, would you mind to include the functional test mentioned in #4291 as well. It wasn't added previously, since `load-plugins` wasn't working.
